pkg/convert/tf.go

```diff
@@ -3236,6 +3250,22 @@ func translateModuleSourceCode(
 
 		body := file.Body()
 
+		// Add package metadata to only the top level module.
+		// This should be true for only the entry point into the conversion, all
+		// recursive module translations should set this false.
+		// This guarantees that the package blocks are only written in one place (eg main.pp)
+		if topLevelModule {
+			for name, provider := range requiredProviders {
+				block, d := getPackageBlock(name, provider)
```

Comment:
I think this needs to parse provider.Source() for the actual name, not just use the local name.

Reply:
well we need the "name" to know what to name the package block, but I think you mean parse it to do the check if it is in pulumiverse or not in the next change?

Otherwise we should keep the name of the block the same as in tf, yes?